docs/architecture/component_templates.md

```diff
@@ -0,0 +1,134 @@
+    { key = "0xDE0B1140012A9FD912F18AD9EC85E40F4CB697AE", value = { name = "value_placeholder", description = "This value will be defined at the moment of instantiation", type = "word" } }
+]
+```
+
+#### Header
+
+The metadata header specifies four fields:
+
+- `name`: The component template's name
+- `description` (optional): A brief description of the component template and its functionality
+- `version`: A semantic version of this component template
+- `supported-types`: Specifies the types of accounts on which the component can be used. Valid values are `FungibleFaucet`, `NonFungibleFaucet`, `RegularAccountUpdatableCode` and `RegularAccountImmutableCode`
+
+#### Storage entries
+
+An account component template can have multiple storage entries. A storage entry can specify either a **single-slot value**, a **multi-slot value**, or a **storage map**.
+
+Each of these storage entries contain the following fields:
+
+- `name`: A name for identifying the storage entry
+- `description` (optional): Describes the intended function of the storage slot within the component definition
+
+Additionally, based on the type of the storage entry, there are specific fields that should be specified.
+
+##### Single-slot value
+
+A single-slot value fits within one slot (i.e., one word).
+
+For a single-slot entry, the following fields are expected:
+
+- `slot`: Specifies the slot index in which the value will be placed
+- `value` (optional): Contains the initial storage value for this slot. Will be interpreted as a `word` unless another `type` is specified
+- `type` (optional): Describes the expected type for the slot
+
+If no `value` is provided, the entry acts as a placeholder, requiring a value to be passed at instantiation. In this case, specifying a `type` is mandatory to ensure the input is correctly parsed.
+Valid types for a single-slot value are `word` or `auth::rpo_falcon512::pub_key`.
+
+Note that at least one of `value` and `type` should be specified. 
+
+In the above example, the first and second storage entries are single-slot values.
+
+##### Multi-slot value
+
+Multi-slot values are composite values that exceed the size of a single slot (i.e., more than one `word`).
+
+For multi-slot values, the following fields are expected:
+
+- `slots`: Specifies the list of slots that the value comprises
+- `values`: Contains the initial storage value for the specified slots
+
+Placeholders can currently not be defined for multi-slot values.
+
+##### Storage maps
+
+Storage maps consist of key-value pairs, where both keys and values are single words.
+
+Storage map entries can specify the following fields:
+
+- `slot`: Specifies the slot index in which the value will be placed
+- `values`: Contains a list of map entries, defined by a `key` and `value `
+
+Storage maps `key`s and `value`s are word-long values, which can be defined as placeholders.
+
+In the example, the third storage entry defines a storage map.
+
+#### Specifying values and their types
```

Comment:
What do you think about moving this section to before `Header`? This serves as a good reminder what felt and word mean in case people are unfamiliar/forgot.

Reply:
Makes sense, moved it